third_party/nvfuser/csrc/ir_internal_nodes.h

```diff
@@ -190,6 +190,10 @@ class TORCH_CUDA_CU_API ScatterOp : public Expr {
   ScatterOpType getScatterOpType() const {
     return attribute(2)->as<Attribute<ScatterOpType>>()->value;
   }
+
+  void updateOutputSelectAxis(Val* mutated) {
```

Comment:
Generally we try to make expressions immutable. Can you just create a new `ScatterOp` with the updated IterDomain instead of modifying an existing `ScatterOp` expr?

Or, it seems we could just have an integer position attribute referring to the select axis. That seems like a simpler design.